packages/hub/src/scope.ts

```diff
@@ -68,19 +76,37 @@ export class Scope {
   /**
    * This will be called after {@link applyToEvent} is finished.
    */
-  protected async notifyEventProcessors(event: SentryEvent, hint?: SentryEventHint): Promise<SentryEvent | null> {
-    let processedEvent: SentryEvent | null = event;
-    for (const processor of [...getGlobalEventProcessors(), ...this.eventProcessors]) {
-      try {
-        processedEvent = await processor({ ...processedEvent }, hint);
-        if (processedEvent === null) {
-          return null;
+  protected notifyEventProcessors(
+    processors: EventProcessor[],
+    event: SentryEvent | null,
+    hint?: SentryEventHint,
+    index: number = 0,
+  ): SyncPromise<SentryEvent | null> {
+    return new SyncPromise<SentryEvent | null>((resolve, reject) => {
+      const processor = processors[index];
+      if (event === null || !isFunction(processor)) {
+        resolve(event);
+      } else {
+        try {
+          const result = processor({ ...event }, hint) as SentryEvent | null;
+          if (isThenable(result)) {
+            (result as Promise<SentryEvent | null>)
+              .then((final: SentryEvent | null) => {
+                this.notifyEventProcessors(processors, final, hint, index + 1)
```

Comment:
Double `catch` can make it hard to track bugs. We already had this issue once with buffers, remember?